custom_components/ha_washdata/ws_api.py

```
@@ -6064,3 +6328,471 @@ def ws_start_playground_cycle_detail(
+        entry_id, "history_import", "Scanning power history",
+        label_key="task.history_import.scanning",
+    )
+    _raw = hass.async_create_task(
+        _history_import_scan_task(hass, task, entry_id, msg["token"])
+    )
+    if _raw is not None:
+        reg.link_asyncio_task(task.id, _raw)
+    _send_result(connection, msg["id"], "start_history_import_scan", {"task_id": task.id})
+
+
+async def _history_import_apply_task(
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)